baton-maven-plugin/src/main/java/org/technologybrewery/baton/config/MigrationTarget.java

```diff
@@ -0,0 +1,76 @@
+package org.technologybrewery.baton.config;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.commons.json.AbstractValidatedElement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Holds target information to support migration execution. A target
+ * represents the implementation details for each migration.
+ */
+public class MigrationTarget extends AbstractValidatedElement {
+
+    @JsonIgnore
+    private static final Logger logger = LoggerFactory.getLogger(MigrationTarget.class);
+
+    @JsonProperty(required = true)
+    private String name;
+
+    @JsonProperty
+    private String description;
+
+    @JsonProperty(required = true)
+    private String implementation;
+
+    @JsonProperty
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    private List<FileSet> fileSets = new ArrayList<>();
+
+    public MigrationTarget() {
+    }
+
+    public String getName() {
```

Comment:
What's the correlation between `MigrationTarget` name/description and `Migration` name/description?  I kind of expected that the name provided in `migrations.json` would map directly to the `Migration` name property, but it doesn't seem like that's the case.  I would have expected that these values would be set on the eventual `Migration` object.

Reply:
Migration is quickly overloaded - I decided that the Java interface was more likely to be used by third party users, so I leveraged the term Migration for it.  Target was just the name I choose to disambiguate the configuration json object, which we only really ever use in the BatonMojo to load migrations.json.